src/main/java/programminglife/model/Segment.java

```diff
@@ -92,98 +79,65 @@ public int getIdentifier() {
         return this.id;
     }
 
-    /**
-     * toString method for the node.
-     * @return the {@link String} representation of a {@link Segment}
-     */
     @Override
-    public String toString() {
-        String sequence = this.getSequence();
-        return String.format("Segment<%d>[s(%d):%s]",
-                this.getIdentifier(),
-                sequence.length(),
-                StringUtils.abbreviate(sequence, 11));
+    public Collection<? extends Edge> getChildEdges() {
+        Collection<Link> result = new HashSet<>();
+        for (Node node : graph.getChildren(this.id)) {
+            result.add(new Link(this, node, graph.getGenomes(node)));
+        }
+        return result;
     }
 
-    /**
-     * Get a {@link XYCoordinate} representing the size of the {@link Segment}.
-     * @return The size of the {@link Segment}
-     */
-    public XYCoordinate getSize() {
-        if (!drawDimensionsUpToDate) {
-            setDrawDimensions();
+    @Override
+    public Collection<? extends Edge> getParentEdges() {
+        Collection<Link> result = new HashSet<>();
```

Comment:
I believe a LinkedHashSet is faster when you only have a small amount of entries.